include/deal.II/sundials/n_vector.templates.h

```diff
@@ -363,7 +363,8 @@ SUNDIALS::internal::unwrap_nvector(N_Vector v)
 {
   Assert(v != nullptr, ExcInternalError());
   Assert(v->content != nullptr, ExcInternalError());
-  auto *pContent = reinterpret_cast<NVectorContent<VectorType> *>(v->content);
+  NVectorContent<VectorType> *pContent =
+    reinterpret_cast<NVectorContent<VectorType> *>(v->content);
```

Comment:
IMHO, the previous version is cleaner. I don't see a point in repeating the type name.
The rest of the changes (to `SUNDIALS::internal::create_empty_nvector()`) look good to me, though.

Reply:
I'm just not a huge fan of `auto` variables in general, and neither are IDEs that then can't properly determine the type of variables. I think this case is probably arguable.

Not sure what to do. Do you want me to revert this part?

Reply:
I also would prefer to use `auto` here so that the type is not written twice - otherwise the patch looks ready to me.

Reply:
I also think that the use of `auto` in this instance would be preferable.

Reply:
I reverted to the use of `auto` upon popular request :-)